File: Api.Server/src/Domain/Admin/Entity/Employee/EmployeeOrgModel.cs
// ------------------------------------------------------------------------
// Apache开源许可证
// 
// 版权所有 © 2018-Now 小方
// 
// 许可授权：
// 本协议授予任何获得本软件及其相关文档（以下简称“软件”）副本的个人或组织。
// 在遵守本协议条款的前提下，享有使用、复制、修改、合并、发布、分发、再许可、销售软件副本的权利：
// 1.所有软件副本或主要部分必须保留本版权声明及本许可协议。
// 2.软件的使用、复制、修改或分发不得违反适用法律或侵犯他人合法权益。
// 3.修改或衍生作品须明确标注原作者及原软件出处。
// 
// 特别声明：
// - 本软件按“原样”提供，不提供任何形式的明示或暗示的保证，包括但不限于对适销性、适用性和非侵权的保证。
// - 在任何情况下，作者或版权持有人均不对因使用或无法使用本软件导致的任何直接或间接损失的责任。
// - 包括但不限于数据丢失、业务中断等情况。
// 
// 免责条款：
// 禁止利用本软件从事危害国家安全、扰乱社会秩序或侵犯他人合法权益等违法活动。
// 对于基于本软件二次开发所引发的任何法律纠纷及责任，作者不承担任何责任。
// ------------------------------------------------------------------------

namespace Fast.Admin.Entity;

/// <summary>
/// <see cref="EmployeeOrgModel"/> 职员机构表Model类
/// </summary>
[SugarTable("EmployeeOrg", "职员机构表")]
[SugarDbType(DatabaseTypeEnum.Admin)]
public class EmployeeOrgModel : IDatabaseEntity
{
    /// <summary>
    /// 职员Id
    /// </summary>
    [SugarColumn(ColumnDescription = "职员Id", IsPrimaryKey = true)]
    public long EmployeeId { get; set; }

    /// <summary>
    /// 机构Id
    /// </summary>
    [SugarColumn(ColumnDescription = "机构Id")]
    public long OrganizationId { get; set; }

    /// <summary>
    /// 部门Id
    /// </summary>
    [SugarColumn(ColumnDescription = "部门Id", IsPrimaryKey = true)]
    public long DepartmentId { get; set; }

    /// <summary>
    /// 是否为主部门
    /// </summary>
    [SugarColumn(ColumnDescription = "是否为主部门")]
    public YesOrNotEnum IsPrimary { get; set; }

    /// <summary>
    /// 职位Id
    /// </summary>
    [SugarColumn(ColumnDescription = "职位Id", IsPrimaryKey = true)]
    public long PositionId { get; set; }

    /// <summary>
    /// 职级Id
    /// </summary>
    [SugarColumn(ColumnDescription = "职级Id")]
    public long? JobLevelId { get; set; }

    /// <summary>
    /// 主管Id
    /// </summary>
    [SugarColumn(ColumnDescription = "主管Id")]
    public long? LeaderEmployeeId { get; set; }
}

File: App.Client/src/api/services/employee/models/EmployeeOrgModel.ts
/**
 * Fast.Admin.Entity.EmployeeOrgModel 职员机构表Model类
 */
export interface EmployeeOrgModel {
  /**
   * 职员Id
   */
  employeeId?: number;
  /**
   * 机构Id
   */
  orgId?: number;
  /**
   * 机构名称
   */
  orgName?: string;
  /**
   * 机构名称
   */
  orgNames?: Array<string>;
  /**
   * 部门Id
   */
  departmentId?: number;
  /**
   * 部门名称
   */
  departmentName?: string;
  /**
   * 部门名称
   */
  departmentNames?: Array<string>;
  /**
   * 是否为主部门
   */
  isPrimary?: boolean;
  /**
   * 职位Id
   */
  positionId?: number;
  /**
   * 职位名称
   */
  positionName?: string;
  /**
   * 职级Id
   */
  jobLevelId?: number;
  /**
   * 职级名称
   */
  jobLevelName?: string;
  /**
   * 是否为负责人
   */
  isPrincipal?: boolean;
}

